### PR TITLE
#34674 Typography - globals update

### PR DESCRIPTION
**Pull request by kruchy8**
https://github.com/SnowdogApps/magento2-alpaca-components/issues/85

### CODE REVIEW

File: components/01-globals/typography/heading/_heading.scss [hunk 1]
```diff
@@ -1,16 +1,16 @@
-$headings-margin                : 0 0 0.5em 0 !default;
-$headings-font-family           : inherit !default;
-$headings-font-weight           : 500 !default;
-$headings-line-height           : 1.1 !default;
-$headings-color                 : inherit !default;
-$headings-text-transform        : none !default;
+$headings-margin        : 0 0 0.5em 0 !default;
+$headings-font-family   : $font-family-base-bold !default;
+$headings-font-weight   : 900 !default;
+$headings-line-height   : 1.1 !default;
```
Comment (anqaka) [inline components/01-globals/typography/heading/_heading.scss]: shouldn't we use our default line-height from project here? Why that one should be smaller?

File: components/01-globals/typography/label/_label.scss [hunk 2]
```diff
@@ -1,10 +1,24 @@
-$label-margin-bottom : 8px !default;
-$label-font-size     : $font-size-small !default;
+$label-margin-top   : 0 !default;
+$label-margin-right : 0 !default;
+$label-margin-bottom: 8px !default;
```
Comment (anqaka) [inline components/01-globals/typography/label/_label.scss]: maybe it'll be good to use here our margin value divided by 2? What do you think?

File: components/01-globals/typography/label/_label.scss [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
+$label-margin-left  : 0 !default;
+$label-font-size    : $font-size-small !default;
+$label-color        : $color-secondary !default;
+$label-font-family  : $font-family-base !default;
+$label-font-weight  : $font-weight-normal !default;
+$label-line-height  : $font-line-height !default;
+$label-font-size    : $font-size-small !default;
```
Comment (anqaka) [inline components/01-globals/typography/label/_label.scss]: this value is already defined in line 5

File: components/01-globals/typography/paragraph/_paragraph.scss [hunk 4]
```diff
@@ -1,9 +1,19 @@
-$paragraph-margin     : 0 0 15px 0 !default;
+$paragraph-color      : $color-secondary !default;
+$paragraph-font-family: $font-family-base !default;
+$paragraph-font-weight: $font-weight-normal !default;
 $paragraph-line-height: $font-line-height !default;
+$paragraph-font-size  : $font-size-base !default;
+$paragraph-margin     : 0 0 15px 0 !default;
```
Comment (anqaka) [inline components/01-globals/typography/paragraph/_paragraph.scss]: as a bottom margin (15px) use variable from our global margin :)

File: components/01-globals/typography/label/_label.scss [hunk 5]
```diff
@@ -1,10 +1,24 @@
...
+    margin: $label-margin-top $label-margin-right $label-margin-bottom $label-margin-left;
     font-size: $label-font-size;
+    color : $label-color;
+    font-family: $label-font-family;
+    font-weight: $label-font-weight;
+    line-height: $label-line-height;
+    transition: $label-transition;
 
     &--hidden {
         display: none;
```
Comment (anqaka) [inline components/01-globals/typography/label/_label.scss]: If we will need hidden label in project we shouldn't hide it with `display:none` b/c of a11y. We should hide it like that
```
position: absolute;
clip: rect(0 0 0 0);
width: 1px;
height: 1px;
margin: -1px;
padding: 0;
overflow: hidden;
border: 0;
```
like in visually-heidden links but without `:focus` and `:active` (not needed here)
if you have some question concerning this, please on chat :)

File: components/01-globals/typography/link/_link.scss [hunk 6]
```diff
@@ -1,17 +1,19 @@
...
+$link-line-height     : $font-line-height !default;
 .link {
     color: $link-color;
     text-decoration: $link-decoration;
     transition: $link-transition;
     font-family: $link-font-family;
     font-weight: $link-font-weight;
+    line-height: $link-line-height;
 
     &:hover {
```
Comment (anqaka) [inline components/01-globals/typography/link/_link.scss]: add also `&:focus` here, please

File: components/01-globals/typography/heading/_heading.scss [hunk 7]
```diff
@@ -1,16 +1,16 @@
-$headings-margin                : 0 0 0.5em 0 !default;
-$headings-font-family           : inherit !default;
-$headings-font-weight           : 500 !default;
-$headings-line-height           : 1.1 !default;
-$headings-color                 : inherit !default;
-$headings-text-transform        : none !default;
+$headings-margin        : 0 0 0.5em 0 !default;
+$headings-font-family   : $font-family-base-bold !default;
+$headings-font-weight   : 900 !default;
```
[the remark this answers is not in the document]
Comment (kruchy8) [reply components/01-globals/typography/heading/_heading.scss]: You're right, good to know. Thanks !

File: components/01-globals/typography/heading/_heading.scss (hunk 1 above)
Comment (kruchy8) [reply components/01-globals/typography/heading/_heading.scss]: My bad, done :)

File: components/01-globals/typography/label/_label.scss (hunk 2 above)
Comment (kruchy8) [reply components/01-globals/typography/label/_label.scss]: There was some discussion about it I guess....
image
what do you think ?

File: components/01-globals/typography/label/_label.scss (hunk 3 above)
Comment (kruchy8) [reply components/01-globals/typography/label/_label.scss]: Thank you :)

File: components/01-globals/typography/link/_link.scss (hunk 6 above)
Comment (kruchy8) [reply components/01-globals/typography/link/_link.scss]: Done

File: components/01-globals/typography/paragraph/_paragraph.scss (hunk 4 above)
Comment (kruchy8) [reply components/01-globals/typography/paragraph/_paragraph.scss]: Done

File: components/01-globals/typography/label/_label.scss (hunk 5 above)
Comment (kruchy8) [reply components/01-globals/typography/label/_label.scss]: Great ! Used it with comment about a11y.

File: components/01-globals/typography/label/_label.scss (hunk 2 above)
Comment (anqaka) [reply components/01-globals/typography/label/_label.scss]: I think that we should keep the vertical spaces unify inside the project. I understand that there can be some specific cases and global variable it's not good everywhere, but in that one, I'd set `$<global-margin> / 2` @Igloczek what do you think, it's already m2 less-bariables-dependencies-tree

File: components/01-globals/typography/heading/_heading.scss (hunk 7 above)
[the remark this answers is not in the document]
Comment (anqaka) [reply components/01-globals/typography/heading/_heading.scss]: Yes, I think it's our case and I agree with you. We set the `font-weight` in [font component](https://github.com/SnowdogApps/magento2-alpaca-components/blob/develop/components/01-globals/_fonts/_fonts.scss), so we need to change inside [globals variable](https://github.com/SnowdogApps/magento2-alpaca-components/blob/develop/components/01-globals/variables/_variables.scss) can you make an adjustments and delete `$font-weight-bold` from variables?  
hmmm indeed I think we can delete all 3 `$font-weight-...` variable or leave only `$font-weight-base` set to normaln... What do you think?

File: components/01-globals/typography/label/_label.scss (hunk 2 above)
Comment (Igloczek) [reply components/01-globals/typography/label/_label.scss]: Good question... I'll suggest keeping value defined directly here and then (as some alpha/beta testing) try to customize some parts of the theme to see if we should change it or not, b/c I'm not sure if the spacing between forms elements should be directly related to i.e. distance between blocks of content.

Comment (kruchy8) [reply components/01-globals/typography/label/_label.scss]: Ok so I leave it this way.

**Pull request merged**